extensions/amp-story-panning-media/0.1/amp-story-panning-media.js

```diff
@@ -32,55 +35,105 @@ export class AmpStoryPanningMedia extends AMP.BaseElement {
+      this.element_.querySelector('amp-img')
+    );
 
-    /** @private {?string} */
-    this.x_ = null;
+    /** @public {?string} Sent to siblings to update their position. */
+    this.x = this.element_.getAttribute('x') || '0%';
 
-    /** @private {?string} */
-    this.y_ = null;
+    /** @public {?string} Sent to siblings to update their position. */
+    this.y = this.element_.getAttribute('y') || '0%';
 
-    /** @private {?string} */
-    this.zoom_ = null;
+    /** @public {?string} Sent to siblings to update their position. */
+    this.zoom = this.element_.getAttribute('zoom') || '1';
+
+    /** @private {Array<Element>} */
+    this.siblings_ = [];
   }
 
   /** @override */
   buildCallback() {
-    this.x_ = this.element_.getAttribute('x') || '0%';
-    this.y_ = this.element_.getAttribute('y') || '0%';
-    this.zoom_ = this.element_.getAttribute('zoom') || '1';
+    this.getSiblings();
+
+    // Initialize all services before proceeding
+    return Promise.all([
+      Services.storyStoreServiceForOrNull(this.win).then((storeService) => {
+        storeService.subscribe(StateProperty.CURRENT_PAGE_ID, (currPageId) => {
+          const isOnActivePage = currPageId === this.getPageId_();
+          if (isOnActivePage) {
+            this.updateSiblings_();
+          }
+        });
+      }),
+    ]).then(() => Promise.resolve());
   }
 
   /** @override */
   layoutCallback() {
-    const ampImgEl = dev().assertElement(
-      this.element_.querySelector('amp-img')
-    );
-    return whenUpgradedToCustomElement(ampImgEl)
-      .then(() => ampImgEl.signals().whenSignal(CommonSignals.LOAD_END))
+    return whenUpgradedToCustomElement(this.ampImgEl_)
+      .then(() => this.ampImgEl_.signals().whenSignal(CommonSignals.LOAD_END))
       .then(() => {
-        this.image_ = dev().assertElement(this.element.querySelector('img'));
+        const imgEl = dev().assertElement(this.element_.querySelector('img'));
         // Remove layout="fill" classes so image is not clipped.
-        this.image_.classList = '';
-        // Fill image to 100% height of viewport.
+        imgEl.classList = '';
+        // Centers the amp-img horizontally. The image does not load if this is done in CSS.
         // TODO(#31515): Handle base zoom of aspect ratio wider than image
-        setStyles(this.image_, {height: '100%'});
-        return this.updatePosition_();
+        setStyles(this.ampImgEl_, {
+          left: 'auto',
+          right: 'auto',
+        });
+        return Promise.resolve();
       })
       .catch(() => user().error(TAG, 'Failed to load the amp-img.'));
   }
 
+  /** @private */
+  getSiblings() {
+    document.querySelectorAll('amp-story-panning-media').forEach((sibling) => {
+      // TODO Group siblings to be transitioned together #31932
+      sibling.getImpl().then((siblingImpl) => {
+        this.siblings_.push(siblingImpl);
+      });
+    });
+  }
+
+  /** @private */
+  updateSiblings_() {
```

Comment:
Would it make sense to only call this on the previous and next pages? Or is there a need to update all the panning medias?

Reply:
The current implementation needs to update all grouped children. 
That breaks the smooth transition when clicking quickly.
if the user clicks through from page 1-3 quickly, the component on page 3 will start transitioning from a different point.

This helps me realize an approach that might be better. If I animate with JS instead of css I could get that value then set the next and previous page's component positions programatically. 🤔